### PR TITLE
[STARTERS] Multindex gen

### PR DESCRIPTION
**Pull request by Nesquickforce**
Multindex generator

### CODE REVIEW

**Comment by khoi-nguyen** [comment]
Do you want me to review this?

**Comment by Nesquickforce** [comment]
Yes please.

File: pandoc/generator.py [hunk 1]
```diff
@@ -1,15 +1,41 @@
...
+        power_1 = randint(1 if level <= 4 else -9, 5 if level == 3 else 9)
+        power_2 = randint(1 if level <= 4 else -9, 5 if level == 3 else 9)
+
+    # y**m * y**n where y can be integer or variable
+    elif level <= 8:
+        cointoss = randint(0, 1)
+        if cointoss:
+            base = randint(2, 9)
+        else:
+            base = choice(['a', 'b', 'c', 'm', 'n', 'x', 'y', 'z'])
```
Comment (khoi-nguyen) [inline pandoc/generator.py:28]: Good try, works well but can be made shorter without if/else. Can be shortened with
    base = choice(['a', 'b', 'c', 'm', 'n', 'x', 'y', 'z'] + range(2, 10))

Again, please try to understand this before copy-pasting it.

File: pandoc/generator.py [hunk 2]
```diff
@@ -1,15 +1,41 @@
...
-    base = randint(1, 9)
-    power_1 = randint(0, 9)
-    power_2 = randint(0, 9)
-    return ('Multiply Index',) + mult(f'a^{power_1}', f'a^{power_2}', a=base)
+    # a**m * a**n where a is an integer
+    if level <= 2:
+        base = randint(2, 5)
+        power_1 = randint(1, 3 if level == 1 else 5)
+        power_2 = randint(1, 3 if level == 1 else 5)
+
```
Comment (khoi-nguyen) [inline pandoc/generator.py:14]: Be consistent with blanklines (think paragraphs).

If you leave a blank line, then you're starting a new idea but as you're still in the if/else thing...

File: pandoc/generator.py [hunk 3]
```diff
@@ -1,15 +1,41 @@
...
+            base = randint(2, 9)
+        else:
+            base = choice(['a', 'b', 'c', 'm', 'n', 'x', 'y', 'z'])
+        power_1 = randint(-9, 9)
+        power_2 = randint(-9, 9)
+    else:
+        base = randint(-5, -2)
+        power_1 = randint(-4, -1)
+        power_2 = randint(-4, 1)
+    if isinstance(base, int):
```
Comment (khoi-nguyen) [inline pandoc/generator.py:35]: It's strange that you separate if/elif with a blank line, but not else with the rest

**Comment by khoi-nguyen** [comment]
Also, give your commits proper name, not "."

**Pull request merged**